### PR TITLE
Lab4

### PR DESCRIPTION
**Pull request by StasDen**
Implemented code of lab4(maven is here)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # repoForJava
 Java code is here
+Updated our project
```
Comment (ChumakAn) [inline README.md]: Add descriptive readme please

File: src/main/java/com/lviv/IoT/AllOfInsurances.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.lviv.IoT;
+
+// A class for our lists
+public class AllOfInsurances {
+
+    public String nameOfInsurance;
+
+    public AllOfInsurances(String nameOfInsurance) {
+
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/AllOfInsurances.java:9]: remove blank line

File: src/main/java/com/lviv/IoT/AllOfInsurances.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        this.nameOfInsurance = nameOfInsurance;
+    }
+
+    // Function to print info in the right way
+    public String toString() {
+
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/AllOfInsurances.java:15]: blank line, format your code

File: src/main/java/com/lviv/IoT/BrokerManager.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.lviv.IoT;
+
+// Inheritance is here
+public class BrokerManager extends Insurances {
+
+    public void printSomeInfoAboutInsurances() {
+
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/BrokerManager.java:7]: same here

File: src/main/java/com/lviv/IoT/InsuranceManager.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+package com.lviv.IoT;
+
+import java.util.*;
+import java.util.List;
+
+// Our class 'Manager' for some sorting functions
+public class InsuranceManager {
+
+    public String[] advancedTypesOfInsurances = {"vehicle", "health", "casualty", "gap", "life", "burial"};
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/InsuranceManager.java]: it's better to use enums

File: src/main/java/com/lviv/IoT/InsuranceManager.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    public void showAdvancedTypesOfInsurances() {
+
+        System.out.println("Our advanced types of insurances: " + Arrays.toString(advancedTypesOfInsurances));
+    }
+
+    // Not all types of insurances are on sale in summer
+    public String findAndShowTypesOfInsurancesOnSaleInSummer() {
+
+        String[] typesOfInsurancesOnSaleInSummer = {"health", "casualty", "gap", "life"};
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/InsuranceManager.java]: better to use enum

File: src/main/java/com/lviv/IoT/InsuranceManager.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+        System.out.println("Our types of insurances on sale in summer: " + Arrays.toString(typesOfInsurancesOnSaleInSummer));
+
+        return Arrays.toString(typesOfInsurancesOnSaleInSummer);
+    }
+
+    public void sortTypesOfInsurances(String[] typesOfInsurances) {
+
+        // Using built-in method to sort our array
+        Arrays.sort(typesOfInsurances);
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/InsuranceManager.java:32]: what about using Streams?

File: src/main/java/com/lviv/IoT/Insurances.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package com.lviv.IoT;
+
+// An abstract class for inheritance
+public abstract class Insurances {
+
+    // Some info about our insurances
+    public String[] typesOfInsurances = {"vehicle", "health", "casualty"};
```
Comment (ChumakAn) [inline src/main/java/com/lviv/IoT/Insurances.java]: better to use enum

File: src/main/resources/META-INF/MANIFEST.MF [hunk 9]
```diff
@@ -0,0 +1,6 @@
+Manifest-Version: 1.0
```
Comment (ChumakAn) [inline src/main/resources/META-INF/MANIFEST.MF]: it's better not to upload this file on git

**Review by ChumakAn — CHANGES_REQUESTED** [review]

**Comment by StasDen** [comment]
Last commit: implemented the code of lab4*

**Pull request merged**